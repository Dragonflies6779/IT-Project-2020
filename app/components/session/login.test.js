var React = require('react');
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import { screen ,getByLabelText,
  getByText,
  getByTestId,
  queryByTestId, } from '@testing-library/dom'
var Login = require('./login.js');

const setup = ( search ) => {
  const utils = render(<Login />)
  const input = utils.getByPlaceholderText( search )
  return {
    input,
    ...utils,
  }
}

it('renders correctly', () => {
  const tree = renderer.create(<Login/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Can input values to Email', () => {
  const { input } = setup('Email Address')
  fireEvent.change(input, {target: {value : 'Abcd'} })
  expect(input.value).toBe('Abcd')
})

it('Can input values to Password', () => {
  const { input } = setup('Password')
  fireEvent.change(input, {target: {value : 'Abcd'} })
  expect(input.value).toBe('Abcd')
})

it('The button is not disabled and works', () => {
  const utils = render(<Login />)
  const input = utils.getByRole('button')
  expect(input.hasAttribute('disabled')).toBe(false)
})
