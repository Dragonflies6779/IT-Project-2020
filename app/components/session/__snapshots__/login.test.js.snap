// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="jumbotron jumbotron-fluid"
>
  <div
    className="WebHeader"
  >
    <div />
    <div
      className="col-md-4"
    />
    <div
      className="col-md-4 margin-top-30"
    >
      <center>
        <h3>
          Login
        </h3>
        <br />
        <div
          className="enter-form"
        >
          <input
            className="form-control"
            onKeyPress={[Function]}
            placeholder="Email Address"
            type="email"
          />
          <br />
          <input
            className="form-control"
            onKeyPress={[Function]}
            placeholder="Password"
            type="password"
          />
          <br />
          <button
            className="btn"
            onClick={[Function]}
          >
            Login
          </button>
          <br />
          <div
            className="linking"
          >
            No account?
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Sign Up!
            </a>
          </div>
        </div>
      </center>
    </div>
    <div
      className="col-md-4"
    />
  </div>
</div>
`;
